opentracing/recorder.go

```diff
@@ -0,0 +1,90 @@
+package opentracing
+
+import (
+	"fmt"
+	"log"
+	"sync"
+
+	basictracer "github.com/opentracing/basictracer-go"
+	"sourcegraph.com/sourcegraph/appdash"
+)
+
+// Recorder implements basictracer.Recorder.
+type Recorder struct {
+	collector appdash.Collector
+	logOnce   sync.Once
+	verbose   bool
+}
+
+// NewRecorder forwards basictracer.RawSpans to an appdash.Collector.
+func NewRecorder(collector appdash.Collector, opts Options) *Recorder {
+	return &Recorder{
+		collector: collector,
+		verbose:   opts.Verbose,
+	}
+}
+
+// RecordSpan converts a RawSpan into the Appdash representation of a span
+// and records it to the underlying collector.
+func (r *Recorder) RecordSpan(sp basictracer.RawSpan) {
+	spanID := appdash.SpanID{
+		Span:   appdash.ID(uint64(sp.SpanID)),
+		Trace:  appdash.ID(uint64(sp.TraceID)),
+		Parent: appdash.ID(uint64(sp.ParentSpanID)),
+	}
+
+	r.collectEvent(spanID, appdash.SpanName(sp.Operation))
+
+	// Record all of the logs. Payloads are thrown out.
+	for _, log := range sp.Logs {
+		r.collectEvent(spanID, appdash.LogWithTimestamp(log.Event, log.Timestamp))
+	}
+
+	for key, value := range sp.Tags {
+		val := []byte(fmt.Sprintf("%v", value))
```

Comment:
Since there is no formal definition in opentracing-go of how tracing systems are to interpret these tags, I think we should at least document how we handle them. I think this would be a good fit for package-level docs in addition to docs on the `RecordSpan` method here.